p2p/src/main/kotlin/maru/p2p/discovery/MaruDiscoveryService.kt

```diff
@@ -80,22 +81,14 @@ class MaruDiscoveryService(
 
   fun getLocalNodeRecord(): NodeRecord = discoverySystem.getLocalNodeRecord()
 
-  init {
-    val discoveryNetworkBuilder = DiscoverySystemBuilder()
-
-    discoveryNetworkBuilder.listen(p2pConfig.ipAddress, p2pConfig.discovery!!.port.toInt())
-    discoveryNetworkBuilder.secretKey(privateKey)
-    discoveryNetworkBuilder.localNodeRecord(localNodeRecord())
-    discoveryNetworkBuilder.bootnodes(bootnodes)
-  }
-
   fun start() {
     discoverySystem
       .start()
       .thenRun {
         this.bootnodeRefreshTask =
```

Comment:
Can you please check the pattern we use in `PeerChainTracker` and `ELSyncService`? We inject `timerFactory` there and there's a testable implementation `TestablePeriodicTimer`, which rules out any inconsistency in the tests

Reply:
done